app/src/main/java/com/napzak/market/data/chat/repositoryimpl/ChatRepositoryImpl.kt

```diff
@@ -0,0 +1,20 @@
+package com.napzak.market.data.chat.repositoryimpl
+
+import com.napzak.market.data.chat.datasource.ChatDataSource
+import com.napzak.market.data.chat.mapper.toDomain
+import com.napzak.market.domain.chat.model.ChatInfo
+import com.napzak.market.domain.chat.repository.ChatRepository
+import javax.inject.Inject
+
+class ChatRepositoryImpl @Inject constructor(
+    private val dataSource: ChatDataSource
+) : ChatRepository {
+    override suspend fun fetchChatInfo(productId: Long): Result<ChatInfo> = runCatching {
+        val response = dataSource.getChatInfo(productId)
+        if (response.status == 200) {
+            response.data.toDomain()
+        } else {
+            throw IllegalStateException("Error fetching chat info: ${response.message}")
+        }
+    }
```

Comment:
p1: 예전 리뷰에서도 언급한 적 있는 것 같은데, dataSource의 함수를 호출했을 때 오류가 발생하면 runCatching에서 자동으로 failure 처리를 합니다! 따라서 다음과 같이 수정해도 별탈 없이 코드가 수행됩니다
```suggestion
        val response = dataSource.getChatInfo(productId)
        response.data.toDomain()
   
```

Reply:
아앗 맞네요!..